test/over_react/component_declaration/builder_integration_tests/component2/required_accessor_integration_test.dart

```diff
@@ -11,152 +11,124 @@
-        render((ComponentTest()
+        expect(() {
+          mount((ComponentTest()
             ..required = null
             ..nullable = true
             ..requiredAndLengthLimited = [1,2]
           )());
-
-        expect(consoleErrors, isNotEmpty, reason: 'should have outputted a warning');
-        expect(consoleErrors, [contains('ComponentTestProps.required')]);
-        expect(consoleErrors, [contains('This Prop is Required for testing purposes.')]);
+        }, logsPropRequiredError('ComponentTestProps.required', 'This Prop is Required for testing purposes.'));
       });
 
       test('on re-render', () {
-        var jacket = mount((ComponentTest()
-            ..required = true
+        TestJacket jacket;
+
+        expect(() {
+          jacket = mount(
+              (ComponentTest()
+                ..required = true
+                ..nullable = true
+                ..requiredAndLengthLimited = [1,2])(),
+              attachedToDocument: true);
+        }, logsNoPropTypeWarnings);
+
+        expect(() {
+          jacket.rerender((ComponentTest()
+            ..required = null
             ..nullable = true
             ..requiredAndLengthLimited = [1,2]
-          )(),
-          attachedToDocument: true,
-        );
-
-        expect(consoleErrors, isEmpty, reason: 'should not have outputted a warning but found: $consoleErrors');
-
-        jacket.rerender((ComponentTest()
-              ..required = null
-              ..nullable = true
-              ..requiredAndLengthLimited = [1,2]
-            )());
-
-        expect(consoleErrors, isNotEmpty, reason: 'should have outputted a warning');
-        expect(consoleErrors, [contains('ComponentTestProps.required')]);
-        expect(consoleErrors, [contains('This Prop is Required for testing purposes.')]);
+          )());
+        }, logsPropRequiredError('ComponentTestProps.required', 'This Prop is Required for testing purposes.'));
       });
     });
 
     group('throwing when a prop is nullable and not set', () {
       test('on mount', () {
-        render((ComponentTest()
-          ..required = true
-          ..requiredAndLengthLimited = [1,2]
-        )());
-        expect(consoleErrors, isNotEmpty, reason: 'should have outputted a warning');
-        expect(consoleErrors, [contains('ComponentTestProps.nullable')]);
-        expect(consoleErrors, [contains('This prop can be set to null!')]);
+        expect(() {
+          render((ComponentTest()
+            ..required = true
+            ..requiredAndLengthLimited = [1,2]
+          )());
+        }, logsPropRequiredError('ComponentTestProps.nullable', 'This prop can be set to null!'));
       });
 
       test('on re-render', () {
-        var jacket = mount((ComponentTest()
+        TestJacket jacket;
+
+        expect(() {
+          jacket = mount(
+              (ComponentTest()
+                ..required = true
+                ..nullable = true
+                ..requiredAndLengthLimited = [1,2])(),
```

Comment:
#nit component formatting
```suggestion
                ..requiredAndLengthLimited = [1,2]
              )(),
```